src/routes/list.nim

```diff
@@ -7,41 +8,43 @@ import ".."/[query, types, redis_cache, api]
 import ../views/[general, timeline, list]
 export getListTimeline, getGraphList
 
-template respList*(list, timeline, vnode: typed) =
+template respList*(list, timeline, title, vnode: typed) =
   if list.id.len == 0:
     resp Http404, showError("List \"" & @"list" & "\" not found", cfg)
 
   let
     html = renderList(vnode, timeline.query, list)
-    rss = "/$1/lists/$2/rss" % [@"name", @"list"]
+    rss = "/i/lists/$1/rss" % [@"id"]
 
-  resp renderMain(html, request, cfg, prefs, rss=rss, banner=list.banner)
+  resp renderMain(html, request, cfg, prefs, titleText=title, rss=rss, banner=list.banner)
 
 proc createListRouter*(cfg: Config) =
   router list:
-    get "/@name/lists/@list/?":
+    get "/@name/lists/@slug/?":
       cond '.' notin @"name"
       cond @"name" != "i"
+      let
+        slug = decodeUrl(@"slug")
+        list = await getCachedList(@"name", slug)
+      if list.id.len == 0:
+        resp Http404
+      redirect("/i/lists/" & list.id)
+
+    get "/i/lists/@id/?":
+      cond '.' notin @"id"
       let
         prefs = cookiePrefs()
-        list = await getCachedList(@"name", @"list")
+        list = await getCachedList(id=(@"id"))
+        title = "@" & list.username & "/" & list.name
         timeline = await getListTimeline(list.id, getCursor())
         vnode = renderTimelineTweets(timeline, prefs, request.path)
-      respList(list, timeline, vnode)
+      respList(list, timeline, title, vnode)
 
-    get "/@name/lists/@list/members":
-      cond '.' notin @"name"
-      cond @"name" != "i"
+    get "/i/lists/@id/members":
+      cond '.' notin @"id"
       let
         prefs = cookiePrefs()
-        list = await getCachedList(@"name", @"list")
+        list = await getCachedList(id=(@"id"))
+        title = "@" & list.username & "/" & list.name
         members = await getListMembers(list, getCursor())
-      respList(list, members, renderTimelineUsers(members, prefs, request.path))
-
-    get "/i/lists/@id/?":
-      cond '.' notin @"id"
-      let list = await getCachedList(id=(@"id"))
-      if list.id.len == 0:
-        resp Http404
-      await cache(list)
```

Comment:
Since cache is saved in previous getCachedList , it's not necessary to do it again here.